### PR TITLE
 Nano: update PyTorch inference key feature doc 

### PR DESCRIPTION
**Pull request by rnwang04**
## Description

update PyTorch inference key feature doc 

### 1. Why the change?

Our PyTorch inference key feature doc  is a little out of date, we need to update it with some recent changes.

### 2. User API changes

No change.

### 3. Summary of the change 
- [x] revise installation part of inference
- [x] revise runtime acceleration part
- [x] add bf16 quantization part
- [x] add `optimize` part

### 4. How to test?
- [x] Document test
https://ruonantetdoc.readthedocs.io/en/update_pytorch_inference/doc/Nano/Overview/pytorch_inference.html

### CODE REVIEW

**Comment by rnwang04** [comment]
@Oscilloscope98 @hoshibara @ACupofAir @Siritao please have a look as well. Please feel free to raise any comments

File: docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md [hunk 1]
```diff
@@ -97,23 +110,37 @@ trainer.predict(ort_model, dataloader)
 # trainer.fit(ort_model, dataloader) # this is illegal
 ```
 
+### TorchScript Acceleration
+The [TorchScript](https://pytorch.org/docs/stable/jit.html) usage is a little different from above two cases. In addition to specifying `accelerator=jit`, you can also set `use_ Ipex=True` to enable the additional acceleration provided by [IPEX (Intel® Extension for PyTorch*)](https://www.intel.com/content/www/us/en/developer/tools/oneapi/extension-for-pytorch.html), we generally recommend the combination of `jit` and `ipex`.The following usage is for TorchScript:
```
Comment (MeouSker77) [inline docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md:114]: `use_ipex=True`

File: docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md [hunk 2]
```diff
@@ -172,7 +193,7 @@ trainer.predict(ov_q_model, dataloader)
 Same as using ONNXRuntime accelerator, it equals to converting the model from PyTorch to OpenVINO firstly and then doing quantization on the converted OpenVINO model:
 ```python
 ov_model = InferenceOptimizer.trace(model, accelerator='openvino', input_sample=x):
```
Comment (MeouSker77) [inline docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md:195]: remove trailing ':'

File: docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md [hunk 3]
```diff
@@ -221,10 +242,111 @@ from torchmetrics.classification import Accuracy
...
+
+### BFloat16 Quantization
+
+BigDL-Nano has support [mixed precision inference](https://pytorch.org/docs/stable/amp.html?highlight=mixed+precision) with BFloat16 and a series of additional performance tricks. BFloat16 Mixed Precison inference combines BFloat16 and FP32 during inference, which could lead to increased performance and reduced memory usage. Compared to FP16 mixed precison, BFloat16 mixed precision has better numerical stability.
+It's quite easy for you use BFloat16 Quantization as below:
+```python
+bf16_model = InferenceOptimizer.quantize(model,
+                                         precision='bf16')
+# run simple prediction with transparent acceleration
+with InferenceOptimizer.get_context():
```
Comment (MeouSker77) [inline docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md:261]: get_context(bf16_model) ?

**Review by MeouSker77 — APPROVED** [review]
others LGTM

File: docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md (hunk 1 above)
Comment (rnwang04) [reply docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md]: done

File: docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md (hunk 2 above)
Comment (rnwang04) [reply docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md]: have removed related part of "trace the quantize".

File: docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md (hunk 3 above)
Comment (rnwang04) [reply docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md]: Done.

**Comment by hoshibara** [comment]
lgtm

**Review by hoshibara — APPROVED** [review]

File: docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 BigDL-Nano provides several APIs which can help users easily apply optimizations on inference pipelines to improve latency and throughput. Currently, performance accelerations are achieved by integrating extra runtimes as inference backend engines or using quantization methods on full-precision trained models to reduce computation during inference. InferenceOptimizer (`bigdl.nano.pytorch.InferenceOptimizer`) provides the APIs for all optimizations that you need for inference.
 
-For runtime acceleration, BigDL-Nano has enabled three kinds of runtime for users in `InferenceOptimizer.trace()`, ONNXRuntime, OpenVINO and jit.
+For runtime acceleration, BigDL-Nano has enabled three kinds of runtime for users in `InferenceOptimizer.trace()`, ONNXRuntime, OpenVINO and TorchScript.
```
Comment (TheaperDeng) [inline docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md:5]: three kinds of graph mode format and corresponding runtime

File: docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md [hunk 5]
```diff
@@ -22,6 +22,27 @@ For quantization, BigDL-Nano provides only post-training quantization in `Infere
```
Comment (TheaperDeng) [inline docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md:22]: In line 14, we may just remove "Model conversion to 16-bit like BF16 is supported now."

**Review by TheaperDeng — COMMENTED** [review]
others LGTM

File: docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md [hunk 6]
```diff
@@ -22,6 +22,27 @@ For quantization, BigDL-Nano provides only post-training quantization in `Infere
...
+    - `INC (Intel Neural Compressor) <https://github.com/intel/neural-compressor>`_: ``pip install neural-compressor``
+
+    - `OpenVINO <https://www.intel.com/content/www/us/en/developer/tools/openvino-toolkit/overview.html>`_: ``pip install openvino-dev``
+
+    - `ONNXRuntime <https://onnxruntime.ai/>`_: ``pip install onnx onnxruntime onnxruntime-extensions onnxsim neural-compressor``
+
+    We recommand installing all dependencies by ``pip install --pre --upgrade bigdl-nano[pytorch,inference]``, because you may run into version issues if you install dependencies manually.
+```
+
 ##  Runtime Acceleration
```
[the remark this answers is not in the document]
Comment (rnwang04) [reply docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md]: Have revised to `Graph Mode Acceleration`

File: docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md (hunk 4 above)
Comment (rnwang04) [reply docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md]: done

File: docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md (hunk 5 above)
Comment (rnwang04) [reply docs/readthedocs/source/doc/Nano/Overview/pytorch_inference.md:22]: Have removed this sentence.

**Pull request merged**